caddytls/handshake.go

```diff
@@ -97,7 +99,27 @@ func (cg configGroup) GetConfigForClient(clientHello *tls.ClientHelloInfo) (*tls
 //
 // This method is safe for use as a tls.Config.GetCertificate callback.
 func (cfg *Config) GetCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	if ClientHelloTelemetry && len(clientHello.SupportedVersions) > 0 {
+		// If no other plugin (such as the HTTP server type) is implementing ClientHello telemetry, we do it.
+		// NOTE: The values in the Go standard lib's ClientHelloInfo aren't guaranteed to be in order.
```

Comment:
Do you have a source for that?

Reply:
Mainly because I'm not in control of that code -- the docs make no such guarantees as far as I know. And the Go standard lib doesn't necessarily use the "raw" values as extracted from the wire; they'll make slight modifications for convenience to most people; but for this case it's necessary to know that the values obtained may not be 100% representative of the how the client sent the data.